### DISCUSSION TITLE
Browser integration with multiple browsers

### DISCUSSION THREAD

**Discussion by parabolapersuader** [keepassxreboot/keepassxc#6747] [Q&A]
I run KeePassXC 2.6.6 from the PPA. And I also run a few variations of Firefox:
- Firefox (from .deb)
- TorBrowser
- Firefox (flatpak)
- LibreWolf (flatpak)

The system install is handled painlessly by KeePassXC. Which I can't say about the others.

There was a way to specify one particular path, which meant choosing a second browser that could have the Browser Integration. Now, when I rechecked the menus for this ticket I notice I can't even find the filed in which I could specify one particular path for the Browser integration. And there is a Tor Browser option, but the app probably expects a system install and looking only for the default paths.

How can I have more than one (it used to be two) browsers connected to the running KeePassXC instance?

**Comment by droidmonkey** [comment]
For each of the non system browsers you will need to manually deploy the native messaging json configuration file. We let you specify one custom browser, beyond that it must be done manually. Consult your browsers documentation for where to place the file.

**Comment by KitchM** [comment]
Please explain the process in more detail.

**Comment by parabolapersuader** [comment]
Maybe a better subject would be ”Better integration with FlatPak”. Maybe the hack is too convoluted, maybe it just doesn't work, but I can't get it to work with KeePassXC and LibreWolf / Firefox in flatpaks. As I would like to migrate to Fedora Silverblue, that would be a very important issue to me, and other SilverBlue users.

**Comment by KitchM** [comment]
I'm not using FlatPac, but the version supplied by Debian 11.  So that would not help for me.

I would be glad to follow a series of steps if someone is just able to create an article describing such.

**Comment by KitchM** [comment]
Does anyone have the exact steps to make other browsers work with the safe?  There used to be an extension for Tor but it does seem to be working any longer.
